logstash-core/lib/logstash/pipeline_settings.rb

```diff
@@ -0,0 +1,35 @@
+# encoding: utf-8
+require "logstash/settings"
+
+module LogStash
+  class PipelineSettings < Settings
+
+    # there are settings that the pipeline uses and can be changed per pipeline instance
+    SETTINGS_WHITE_LIST = [
+      "pipeline.id", "pipeline.system", "pipeline.reloadable",
+      "pipeline.workers", "pipeline.output.workers",
+      "pipeline.batch.size", "pipeline.batch.delay",
+      "config.string", "path.config", "metric.collect", "config.debug",
+      "queue.max_events", "queue.checkpoint.acks", "queue.checkpoint.writes", "queue.checkpoint.interval",
+      "queue.type", "queue.page_capacity", "queue.max_bytes", "path.queue", "queue.drain",
+      "dead_letter_queue.enable",
+      "config.reload.automatic", "config.reload.interval"
+    ]
```

Comment:
We have a lot of whitelisted option, I think we should have 1 option per line. We can easily sort it and keep it sorted that way.